File: src/features/typing/data/go/longest_substring_without_repeating_characters.go
package main

func lengthOfLongestSubstring(s string) int {
	last := make(map[byte]int)
	start := 0
	best := 0
	for i := 0; i < len(s); i++ {
		c := s[i]
		if j, ok := last[c]; ok && j >= start {
			start = j + 1
		}
		last[c] = i
		if i-start+1 > best {
			best = i - start + 1
		}
	}
	return best
}

File: src/features/typing/data/ts/longest_substring_without_repeating_characters.ts
function lengthOfLongestSubstring(s: string): number {
  const last = new Map<string, number>();
  let start = 0;
  let best = 0;
  for (let i = 0; i < s.length; i++) {
    const c = s[i];
    const prev = last.get(c);
    if (prev !== undefined && prev >= start) {
      start = prev + 1;
    }
    last.set(c, i);
    best = Math.max(best, i - start + 1);
  }
  return best;
}
